internal/controllers/clusterrequest_controller.go

```diff
@@ -273,7 +282,10 @@ func (t *clusterRequestReconcilerTask) checkClusterDeployConfigState(ctx context
 	nodePool := &hwv1alpha1.NodePool{}
 	nodePool.SetName(t.object.Status.NodePoolRef.Name)
 	nodePool.SetNamespace(t.object.Status.NodePoolRef.Namespace)
-	t.checkNodePoolProvisionStatus(ctx, nodePool)
+	_, _, err := t.checkNodePoolProvisionStatus(ctx, nodePool)
+	if err != nil {
+		return requeueWithError(err)
+	}
```

Comment:
Should this be applied here as well? Skip checking the rest of other resources status if it times out or fails and periodically check if hw provision is not done yet
```
	if timedOutOrFailed {
		// Timeout occurred or failed, stop requeuing
		return doNotRequeue(), nil
	}
	if !provisioned {
	   requeue
	}
```

Reply:
This check will enforce the use of the test plugin for everyone. Are we sure we want to do that now?                                                                   ```
if !provisioned {
	   requeue
	}
```

Reply:
I think the workaround check has to be added as well. 
Even though we’re watching for NodePool status updates, since a timeout has been added, we’ll need to periodically check.

Reply:
Updated.